rfcs/0009-code-review.md

```diff
@@ -0,0 +1,173 @@
+out considerations for reviewing code at CDS.
+
+## Motivation
+A shared set of considerations helps ease both the burden of being
+reviewed and that of being the reviewer. Code review is inherently
+subjective, different reviewers focus on different things. This can make
+the review system seem opaque. Shared considerations help to ease that
+subjectivity by giving guide rails to make reviews more consistent.
+
+## Proposal
+Below is a list of considerations these are divided into principles and
+technical consideration. Focus on the principles and then lead into the
+technical considerations.
+
+### General guidelines
+ - Code reviews are required at CDS.
+   - Each review should have a technical reviewer.
+   - Add Design, Research, Product Management, Service Owner, as needed.
+ - Tend towards questions rather then statements.
+   - The reviewer may not understand all the considerations a reviewee
+     put into the code.
+   - The reviewee may not understand the reviewers past experience with
+     the code.
+ - [Be nice](https://qz.com/work/625870/after-years-of-intensive-analysis-google-discovers-the-key-to-good-teamwork-is-being-nice/)
+   - Read the CDS [Code of Conduct](https://github.com/cds-snc/.github/blob/master/CODE_OF_CONDUCT.md)
+ - Escalate if the review is blocked.
+   - The goal of the review is that the code looks good to the reviewer.
+     It doesn't have to be perfect, it doesn't have to be how the
+     reviwere would write it. Just, good.
+ - In general authors should merge their own PRs.
+   - This allows them to make any final touch ups or edit the commit
+     message.
+ - Cleanup branches after merge. If possible, enable the auto-delete
+   after merge option on repositories.
+ - Prefer to `Squash and merge` commits.
+   - The development history is in the PR if desired, but typically
+     doesn't add much to the main repository.
+
+
+### Principles for reviewees
+ - The title and description should provide all the needed context.
+   - The PR will serve as a record of the change into the future.
+   - Anything important for the reviewer to know (other approaches tried,
+     testing you’ve done, etc) is worth including.
+   - Link to any associated issues or Trello cards.
+   - If the PR is broken apart to make review easier, explain how it was
+     broken apart and include the next steps in the PR description.
+   - List any steps, caveats or gotchas that the reviewer will need to
+     know.
+ - Do your best to address all comments (even just to say "Done" or put
+   in a thumbs up).
+   - Marking the comments as `Resolved` makes it easier to understand
+     what still needs to be done.
+ - Assume the best intention from review comments
+ - Wait for the tests to pass before you merge.
+ - Tend towards smaller changes. They are easier to understand and
+   faster to review.
+
+
+### Principles for reviewers
+ - Try to comment on the PR in as short an interval is possible (<24
```

Comment:
24 Business hours is essentially a little over 3 business days.  I would wonder if this timeframe should be shortened especially if the comment isn't necessarily a review but when they'll be able to review.

Reply:
I phrased that badly. I was trying to say 24 hrs, but excluding weekends. I'll try to rephrase so it's correct.

Reply:
Better?

Reply:
:)